src/jit/codegenxarch.cpp

```diff
@@ -188,7 +224,13 @@ void                CodeGen::genEmitGSCookieCheck(bool pushReg)
         if (compiler->lvaKeepAliveAndReportThis() && compiler->lvaTable[compiler->info.compThisArg].lvRegister &&
             (compiler->lvaTable[compiler->info.compThisArg].lvRegNum == REG_ECX))
         {
+#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
+            // On System V RDX is used as a second return register for a struct.
+            // Use a diiferent callee-trashed register, so the return value will not get trashed.
+            regGSCheck = REG_RSI;
+#else // !FEATURE_UNIX_AMD64_STRUCT_PASSING
             regGSCheck = REG_RDX;
+#endif // !FEATURE_UNIX_AMD64_STRUCT_PASSING
```

Comment:
We can replace lines 227-233 with

regGSCheck = REG_ARG_1;

Also we should replace REG_ECX with REG_ARG_0.